File: React.UI/ClientApp/src/models/Docente.ts
import {aulaSchema} from "./Aula";
import { z } from "zod";
import {numeric, text, id } from "./Schema";

const docenteSchema = z.object({
  idDocente: id,
  nombres: text,
  apellidoPaterno: text,
  apellidoMaterno: text,
  email: text,
  password: text,
  activo: z.boolean().optional().default(true),
  administrador: z.boolean(),
  aulaAsignada: numeric,
  aulaAsignadaNavigation: aulaSchema,
});

type IDocente = z.TypeOf<typeof docenteSchema>;

export type { IDocente };
export { docenteSchema };

File: AcademicAPI/Models/Kinder/Docente.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace AcademicAPI.Models.Kinder
{
    public partial class Docente
    {
        public int IdDocente { get; set; }
        public string Nombres { get; set; }
        public string ApellidoPaterno { get; set; }
        public string ApellidoMaterno { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public bool? Activo { get; set; }
        public bool? Administrador { get; set; }
        public int AulaAsignada { get; set; }

        public virtual Aula AulaAsignadaNavigation { get; set; }
    }
}
